docs/integrations/webhooks/debugging-webhooks.mdx

```diff
@@ -0,0 +1,93 @@
+---
+title: Debugging webhooks
+description: Understand how to debug your webhooks while developing your application
+---
+
+Developing with webhooks can be a new experience for developers. It can be hard to debug when something goes awry. This guide will cover the basics for debugging and help you to direct your attention to the correct spot.
+
+## Understanding webhooks and local development
+
+When you or a user of your application performs certain actions, a webhook can be triggered. You can see the full list of [webhook events](/docs/integrations/webhooks/overview#supported-webhook-events) to see the various actions that could result in a Webhook. Depending on the events subscribed to in the [Webhook section](https://dashboard.clerk.com/last-active?path=webhooks) of your dashboard, a webhook will be triggered.
+
+When you are developing on your localhost, your application is not internet facing. It can't receive the webhook request. You will need to use a tool that creates a tunnel from the internet to your localhost. These tools will provide temporary or permanant URLs depending on the tool and the plan you subscribe to. Popular tools included `ngrok`, `localtunnel` and `Cloudflare Tunnel`.
+
+Below is an image demonstrating the flow from when you perform an action that triggers a webhook through until the webhook is received by your application
+
+FLOW CHART IMAGE HERE
+
+
+## Debugging
+
+With the Webhook flow in mind, the following areas could cause problems when trying to develop with webhooks:
```

Comment:
You could drop the "Debugging" H2 here, and just close off the previous section with "The rest of this guide will cover the most common causes of webhook errors." or something along those lines. It feels like this section is recreating the ToC